src/utils.js

```diff
@@ -89,6 +89,85 @@ function clearNode(node: Node) {
     setTextContent(node, "");
 }
 
+type BaseElem = {|
+    type: "mathord",
+|} | {|
+    type: "textord",
+|} | {|
+    type: "bin",
+|} | {|
+    type: "rel",
+|} | {|
+    type: "inner",
+|} | {|
+    type: "open",
+|} | {|
+    type: "close",
+|} | {|
+    type: "punct",
+|};
+
+type Group = {|
+    type: "ordgroup",
+    value: Group[],
+|} | {|
+    type: "color",
+    value: {|
+        value: Group[],
+    |},
+|} | {|
+    type: "font",
+    value: {|
+        body: Group,
+    |},
+|} | BaseElem;
+
+/**
+ * Sometimes we want to pull out the innermost element of a group. In most
+ * cases, this will just be the group itself, but when ordgroups and colors have
+ * a single element, we want to pull that out.
+ */
+const getBaseElem = function(group: Group) {
```

Comment:
What's the type of the return value? I'm seen at least 2 different types being returned.